www/nicotine-plus/Portfile

```diff
@@ -15,12 +15,16 @@ description         Nicotine+ is a graphical client for the Soulseek peer-to-pee
 long_description    Nicotine+ aims to be a lightweight, pleasant, free and open-source alternative \
                     to the official Soulseek client, while also providing a comprehensive set of features.
 homepage            https://nicotine-plus.org
-checksums           rmd160  28b4921acb592df0350980d3e3c3fc82d78bddc5 \
-                    sha256  c94bd7ecf6a107445836df80efed9b8c18cab307d812c2b91be18e93dfb4ac1f \
-                    size    3503858
+checksums           rmd160  e144e61dfacaf9ee97a0b85195aefd44d7c7488c \
+                    sha256  1399e8edad1212ca17fdf4a44ab9975c195a57b269c16194e4f06a27d9d3d31a \
+                    size    3508858
 github.tarball_from archive
 
-python.default_version 311
+if {[variant_isset gtk3]} {
+    python.default_version  311
```

Comment:
What's the reason for using Python 3.11 in this case?

Reply:
Perhaps makes sense to peg an old GTK to some Python, assuming that it will be only used on some older systems and optionally.

The default is set to GTK4 and Python 3.12.

Reply:
does this work with Python 3.12; if so, that should be used.

Reply:
For non-default variants it should not.

Reply:
it should

Reply:
@reneeotten We can close this PR, and keep the port outdated, if you prefer.

I am not willing to blindly move everything to Python 3.12, which has issues, for no good reason.

This PR already makes Python 3.12 with gtk4 a default option for all systems. If someone wants an old gtk3 and picks it manually, only then Python 3.11 is picked.

Is this still does not satisfy your preference, I cannot do much further here.
(Well, I could make an explicit non-default variant for Python 3.11, perhaps, if that works better.)

Reply:
I have set gtk4 and python 3.12 as default variants now and added conflicts.